[offload] `rtl.cpp` fails to build with GCC: `ErrorReporting.h:93:3: error: ‘format’ attribute argument 3 value ‘3’ does not refer to a variable argument list`

While testing standalone builds, I've also noticed that `rtl.cpp` does not compile with GCC 14.2.1:

```
[1/9] Building CXX object plugins-nextgen/common/CMakeFiles/PluginCommon.dir/src/PluginInterface.cpp.o
FAILED: plugins-nextgen/common/CMakeFiles/PluginCommon.dir/src/PluginInterface.cpp.o 
/usr/bin/c++ -DDEBUG_PREFIX=\"PluginInterface\" -DLIBC_NAMESPACE=__llvm_libc_common_utils -DOMPT_SUPPORT=1 -DTARGET_NAME=\"PluginInterface\" -I/usr/lib/llvm/20/include -I/home/mgorny/git/llvm-project/offload/plugins-nextgen/common/include -I/home/mgorny/git/llvm-project/offload/build/include -I/home/mgorny/git/llvm-project/offload/include -I/home/mgorny/git/llvm-project/runtimes/cmake/Modules/../../../libc -O3 -DNDEBUG -std=c++17 -fPIC -fno-exceptions -MD -MT plugins-nextgen/common/CMakeFiles/PluginCommon.dir/src/PluginInterface.cpp.o -MF plugins-nextgen/common/CMakeFiles/PluginCommon.dir/src/PluginInterface.cpp.o.d -o plugins-nextgen/common/CMakeFiles/PluginCommon.dir/src/PluginInterface.cpp.o -c /home/mgorny/git/llvm-project/offload/plugins-nextgen/common/src/PluginInterface.cpp
In file included from /home/mgorny/git/llvm-project/offload/plugins-nextgen/common/src/PluginInterface.cpp:17:
/home/mgorny/git/llvm-project/offload/plugins-nextgen/common/include/ErrorReporting.h: In substitution of ‘template<class ... ArgsTy> static void llvm::omp::target::plugin::ErrorReporter::print(ColorTy, const char*, ArgsTy&& ...) [with ArgsTy = {const char* const&}]’:
/home/mgorny/git/llvm-project/offload/plugins-nextgen/common/include/ErrorReporting.h:104:10:   required from here
  104 |     print(BoldRed, "%s", ErrorBanner);
      |     ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mgorny/git/llvm-project/offload/plugins-nextgen/common/include/ErrorReporting.h:93:3: error: ‘format’ attribute argument 3 value ‘3’ does not refer to a variable argument list
   93 |   print(ColorTy Color, const char *Format, ArgsTy &&...Args) {
      |   ^~~~~
/home/mgorny/git/llvm-project/offload/plugins-nextgen/common/include/ErrorReporting.h: In substitution of ‘template<class ... ArgsTy> static void llvm::omp::target::plugin::ErrorReporter::reportError(const char*, ArgsTy&& ...) [with ArgsTy = {const char*&}]’:
/home/mgorny/git/llvm-project/offload/plugins-nextgen/common/include/ErrorReporting.h:110:57:   required from here
  110 |   static void reportError(const char *Str) { reportError("%s", Str); }
      |                                              ~~~~~~~~~~~^~~~~~~~~~~
/home/mgorny/git/llvm-project/offload/plugins-nextgen/common/include/ErrorReporting.h:102:49: error: ‘format’ attribute argument 3 value ‘2’ does not refer to a variable argument list
  102 |   [[gnu::format(__printf__, 1, 2)]] static void reportError(const char *Format,
      |                                                 ^~~~~~~~~~~
/home/mgorny/git/llvm-project/offload/plugins-nextgen/common/include/ErrorReporting.h: In substitution of ‘template<class ... ArgsTy> static void llvm::omp::target::plugin::ErrorReporter::print(const char*, ArgsTy&& ...) [with ArgsTy = {const char*&}]’:
/home/mgorny/git/llvm-project/offload/plugins-nextgen/common/include/ErrorReporting.h:111:45:   required from here
  111 |   static void print(const char *Str) { print("%s", Str); }
      |                                        ~~~~~^~~~~~~~~~~
[…]
```

Mostly wondering if it's a problem with the code or limitation of GCC.

That's against e0ed0333f0fed2e73f805afd58b61176a87aa3ad. To reproduce (sample using Gentoo paths):

```
cmake ${llvm_checkout}/offload -G Ninja -DFFI_INCLUDE_DIR=/usr/lib64/libffi/include -DFFI_LIBRARY_DIR=/usr/lib64
ninja
```

--- a/offload/plugins-nextgen/common/include/ErrorReporting.h
+++ b/offload/plugins-nextgen/common/include/ErrorReporting.h
@@ -80,8 +80,10 @@ class ErrorReporter {
   /// Print \p Format, instantiated with \p Args to stderr.
   /// TODO: Allow redirection into a file stream.
   template <typename... ArgsTy>
-  [[gnu::format(__printf__, 1, 2)]] static void print(const char *Format,
-                                                      ArgsTy &&...Args) {
+#ifdef __clang__ // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=77958
+  [[gnu::format(__printf__, 1, 2)]]
+#endif
+  static void print(const char *Format, ArgsTy &&...Args) {
     raw_fd_ostream OS(STDERR_FILENO, false);
     OS << llvm::format(Format, Args...);
   }
@@ -89,8 +91,10 @@ class ErrorReporter {
   /// Print \p Format, instantiated with \p Args to stderr, but colored.
   /// TODO: Allow redirection into a file stream.
   template <typename... ArgsTy>
-  [[gnu::format(__printf__, 2, 3)]] static void
-  print(ColorTy Color, const char *Format, ArgsTy &&...Args) {
+#ifdef __clang__ // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=77958
+  [[gnu::format(__printf__, 2, 3)]]
+#endif
+  static void print(ColorTy Color, const char *Format, ArgsTy &&...Args) {
     raw_fd_ostream OS(STDERR_FILENO, false);
     WithColor(OS, HighlightColor(Color)) << llvm::format(Format, Args...);
   }
@@ -99,8 +103,10 @@ class ErrorReporter {
   /// a banner.
   /// TODO: Allow redirection into a file stream.
   template <typename... ArgsTy>
-  [[gnu::format(__printf__, 1, 2)]] static void reportError(const char *Format,
-                                                            ArgsTy &&...Args) {
+#ifdef __clang__ // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=77958
+  [[gnu::format(__printf__, 1, 2)]]
+#endif
+  static void reportError(const char *Format, ArgsTy &&...Args) {
     print(BoldRed, "%s", ErrorBanner);
     print(BoldRed, Format, Args...);
     print("\n");

